### DISCUSSION TITLE
Private casts

### DISCUSSION THREAD

**Discussion by codercody** [farcasterxyz/protocol#135] [FIP Stage 1: Ideas]
From a conversation:
> **heheshamona**
> Hi, I’m having trouble understanding privacy. Is there a way to make posts that are only visible to certain sets of people? Or even simpler, is there a way I can associate some kind of secret to my account and then make it so I can authenticate (“log in”) and see that secret?
> **varunsrin**
> long term we'd like to incorporate private messaging at the protocol layer, but its a non trivial amount of work and needs a lot more r&d
> right now, only public messaging is supported

Could we add support for private casts? My particular use case is I want to be able to make a post on my profile that only I can see, but then I want to be able to share my posts on my public profile. That way, if someone visits my profile, they only see the posts I've chosen to show publicly to the world.

Relevant issue: #5

**Comment by varunsrin** [comment]
> My particular use case is I want to be able to make a post on my profile

Posting things that only you can see is a niche feature and encryption is very complex to implement. Unlikely to prioritize this in the near future.
